### PR TITLE
validator: check duplicate key cols when create table

### LINKED ISSUE #1939: Constraints' keys can't be duplicated
1.What did you do?
create table t(a int, b int, key (a, b, a));

2.What did you expect to see?
ERROR 1060 (42S21): Duplicate column name 'a'

3.What did you see instead?
Query OK, 0 rows affected (0.01 sec)

### PR DESCRIPTION
**Pull request by XuHuaiyu**
fix #1939 
This commit check duplicate cols in constraints' key when create table.
PTAL @shenli @coocood @hanfei1991 @zimulala

### CODE REVIEW

File: plan/validator.go [hunk 1]
```diff
@@ -196,6 +196,17 @@ func (v *validator) checkCreateTableGrammar(stmt *ast.CreateTableStmt) {
 			return
 		}
 	}
+	for _, constraint := range stmt.Constraints {
+		switch tp := constraint.Tp; tp {
+		case ast.ConstraintKey:
```
Comment (shenli) [inline plan/validator.go]: Should we check the following constrains:
	ConstraintPrimaryKey
	ConstraintKey
	ConstraintIndex
	ConstraintUniq
	ConstraintUniqKey
	ConstraintUniqIndex

Comment (XuHuaiyu) [inline plan/validator.go]: I'll improve the validator of `create table` in next PR.

**Comment by hanfei1991** [comment]
LGTM

File: ddl/ddl.go [hunk 2]
```diff
@@ -1206,7 +1207,7 @@ func getAnonymousIndex(t table.Table, colName model.CIStr) model.CIStr {
 }
 
 func (d *ddl) CreateIndex(ctx context.Context, ti ast.Ident, unique bool, indexName model.CIStr, idxColNames []*ast.IndexColName) error {
-	err := checkDuplicateColumnName(idxColNames)
+	err := plan.CheckDuplicateColumnName(idxColNames)
```
Comment (coocood) [inline ddl/ddl.go]: Can we move this check to plan validator?

Comment (coocood) [inline ddl/ddl.go]: Oh, it is already checked in plan validator, so this can be removed?

Comment (coocood) [inline ddl/ddl.go]: We can add `AlterTable` check in validator.

**Comment by XuHuaiyu** [comment]
@coocood PTAL

File: plan/validator_test.go [hunk 3]
```diff
@@ -54,6 +54,7 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 		{"create table t(id int auto_increment) ENGINE=MYISAM", true, nil},
 		{"create table t(a int primary key, b int, c varchar(10), d char(256));", true, errors.New("Column length too big for column 'd' (max = 255); use BLOB or TEXT instead")},
 		{"create index ib on t(b,a,b);", true, errors.New("[schema:1060]Duplicate column name 'b'")},
+		{"create table t (a int, b int, key(a, b, A))", true, errors.New("[schema:1060]Duplicate column name 'A'")},
```
Comment (coocood) [inline plan/validator_test.go]: Add test case for alter table add index.

**Comment by coocood** [comment]
LGTM

**Pull request merged**